astropy/units/core.py

```diff
@@ -1487,7 +1488,7 @@ def sort_results(results):
             )
         )
 
-    def to_system(self, system):
+    def to_system(self, system: ModuleType) -> list[CompositeUnit]:
```

Comment:
`ModuleType` seems a bit restrictive here, since it really only needs to have appropriate attributes, but maybe worry about that when a problem arises.

Reply:
The annotation matches the docstring.

Reply:
I think `gala.UnitSystem` is supported by this function, in which case there's real-world code that depends on an expanded definition.

Reply:
It would indeed be best to use a `Protocol` here, but I'm not familiar enough with the code to write it myself and we shouldn't trust the docstring (if we could trust it then `ModuleType` would be correct and a `Protocol` wouldn't be needed). I've removed the annotations from this function entirely, for now.

Reply:
SGTM.